### DISCUSSION TITLE
[Question] Is Jamulus working on Mac M1?

### DISCUSSION THREAD

**Discussion by gegeweb** [jamulussoftware/jamulus#1034] [Technical Support & Help]
Im working on the french translation of the website documentation and to be up to date I need to know if Jamulus have been succefully compiled and tested on macOS Big Sure running on a M1 processor.
If not we'll have to add a warning on the related pages. (see https://github.com/jamulussoftware/jamuluswebsite/issues/88) 

About Mac OS Big Sure, actual binary works fine (Intel processor) and I've succefully compiled Jamulus with QT5 on this OS.
_(instructions on the doc are incomplete (we will fix that), and there is no icon for the `Jamulus.app` application builded, and only client is builded, not the server application)_

I have only some minor cosmetic issues in the UI of the server with the official binary downloaded on sourceforge. (I'll open a new issue later for that).

**Comment by gegeweb** [comment]
Unfortunatly I don't have a Mac M1 (and not sure if I want one… I will probably switch to PC and GNU/Linux in futur) for testing this point.

**Comment by pljones** [comment]
> there is no icon for the Jamulus.app application builded
> and only client is builded, not the server application

Both these points sound like you didn't run the installer build, only the code build.  There is only one executable created.

**Comment by gegeweb** [comment]
> Both these points sound like you didn't run the installer build, only the code build. There is only one executable created.

As I say, probably because documentation is incomplete at this time I've followed the verry succint instructions about macOS here:
https://jamulus.io/wiki/Compiling

* missing how-to installing homebrew (Requirements)
* missing how-to fix the failure with `brew link qt5 --force` if `/usr/local/Frameworks`doesn't exist yet.
* missing the installer build step ;)

And perhaps other things, that was the first time I compiled a software with XCode on macOS (I was curious)…

**Comment by gegeweb** [comment]
> because documentation is incomplete

But maybe it's a deliberate choice?

**Comment by gegeweb** [comment]
> Both these points sound like you didn't run the installer build, only the code build. There is only one executable created.

I've looked on the Apple's doc about XCode but don't find the command for the missing step.
Could you tell me how to do?

Then we will also be able to update the doc.

Thanks.

**Comment by pljones** [comment]
You'll need someone who knows about MacOS, I think.

**Comment by gegeweb** [comment]
> You'll need someone who knows about MacOS, I think.

There is `mac/deploy_mac.sh`wich do the job.

**Comment by ann0see** [comment]
I assume we could compile a native binary for M1 Macs with Xcode 12 somehow. It might even do it automatically (but I don't think so)

**Comment by gegeweb** [comment]
@ann0see 

> I assume we could compile a native binary for M1 Macs with Xcode 12 somehow. It might even do it automatically (but I don't think so)

I unfortunately haven't a Mac M1 to try that.

I'm on Big Sure, I'll check if there is M1 as a target (I am not familiar with XCode).
I've some issues with `deploy_mac.sh` to understand/fix before…

There is a warning on Big Sure that say 11.0 is not a supported configuration for QT, but it compile and works.

**Comment by ann0see** [comment]
You can open the .xcodeproject file probably to change the build options

**Comment by gegeweb** [comment]
> I've some issues with `deploy_mac.sh` to understand/fix before…

https://github.com/al45tair/dmgbuild/issues/29

**Comment by gilgongo** [comment]
Possible issue with Big Sur? https://sourceforge.net/p/llcon/discussion/software/thread/487c151a57/#835c/0af3/37ea/a572

**Comment by gegeweb** [comment]
Jamulus works fine here on Big Sure (Intel processor).

<img width="784" alt="Capture d’écran 2020-12-09 à 09 10 58" src="https://user-images.githubusercontent.com/9108457/101602478-7c034e80-39fe-11eb-9f87-96b925977e6d.png">

<img width="766" alt="Capture d’écran 2020-12-09 à 09 12 26" src="https://user-images.githubusercontent.com/9108457/101602666-c08eea00-39fe-11eb-8993-ab3c5348679f.png">

**Comment by gilgongo** [comment]
OK. So can we close this issue then or do you mean whether we need to know about it working on M1?

**Comment by gegeweb** [comment]
Yes, we need to know if it works on a M1 processor.
We need user feedback for that.
No Mac M1 for testing on my side.

**Reply by mvoitz** [reply]
I am happy to test and do anything to get it work. I have a brand new M1 Mini and 3.7.0 fresh installed. I get an audio error message.

**Reply by njc235** [reply]
What is the error message?

**Reply by mvoitz** [reply]
![image](https://user-images.githubusercontent.com/83690768/117105051-86dbc880-ad7d-11eb-8c4d-6c4eed09750b.png)

**Reply by mvoitz** [reply]
I have reinstalled MacOS to be sure I do not have a bad system. Then I installed Jamulus and Rosetta again. Still this error comes up. I can commit to test everything you need! Happy to Beta Test! We using Jamulus with 20 people minimum usually and scaled up to 100 already! So we are big supporter!!

**Comment by gegeweb** [comment]
Two issues here:
* the binary compiled on a Intel processor work on a M1 processor? (with compatibility mode?)
* If no, can we compile on M1/Big Sure (probably yes) and made the package/dmg? (see above for the issue with dmgbuild on Big Sure / XCode 12)

**Comment by 2xbass** [comment]
I am running 3.6.2 on Big Sur on a M1 MacBook Air and also on one of the old MacBook 12" running Big Sur.  Everything seems fine.

**Reply by mvoitz** [reply]
I have 3.7.0 and it does not work running with Rosetta. Some else any experience?

**Reply by njc235** [reply]
@mvoitz you have to have Rosetta installed

**Reply by mvoitz** [reply]
Hi, yes it installed Rosetta. And does not work. Then I deleted entire iOS and reinstalled only Jamulus, Rosetta install info came up, I agreed and started but it still not work.

**Reply by njc235** [reply]
Is it the security error you are seeing or something else.

**Reply by mvoitz** [reply]
I uploaded the image in the other thread. I checked the Security and Privacy there is no request from iOS to accept the Jamulus App.

**Comment by ann0see** [comment]
I assume it runs with Rosetta 2, not natively?

**Comment by gilgongo** [comment]
Not sure if this is a Big Sur issue or not: https://sourceforge.net/p/llcon/discussion/533517/thread/1b2a9e6078/#8161

**Comment by njc235** [comment]
@ann0see I have it running on my M1 MacMini on BigSur but as you rightly say, it's running I Intel mode via Rosetta 2. Latency does seem a little higher than I would have expected and it may be for that reason. I would have no idea how to compile it on the m1 but I'd be willing to try if anyone wants to show me the way.

**Reply by mvoitz** [reply]
Hi, 3.7.0 with Rosetta does not work on my fresh installed BigSur M1 Mac. Some has some help? I am happy to test!

**Comment by ann0see** [comment]
You could try https://jamulus.io/wiki/Compiling

or even better run mac/deploy_mac.sh and open the Xcode project file with xcode

**Comment by ann0see** [comment]
I think Qt is working on native arm64 support: https://bugreports.qt.io/plugins/servlet/mobile#issue/QTBUG-85279

**Comment by gegeweb** [comment]
> You could try https://jamulus.io/wiki/Compiling
> 
> or even better run mac/deploy_mac.sh and open the Xcode project file with xcode

I've succefully compiled Jamulus with XCode on Bug Sur (Intel processor) following these, but there is an issue with the deploy script.
See my previous comment : https://github.com/corrados/jamulus/issues/749#issuecomment-739247428
There is a warning about unsuported platform with QT, but it works (compile) on Intel processor.

**Comment by corrados** [comment]
Has anybody been able to compile Jamulus to run native ARM code on the new Mac?

**Comment by njc235** [comment]
> Has anybody been able to compile Jamulus to run native ARM code on the new Mac?

I have an M1 MacMini but I don't have the first idea how to compile code. If anyone has the time to spend with me, I am happy for my machine to try to do this.

**Comment by gilgongo** [comment]
Hi all - until we have some definable action to take to add to a to-do list or codebase change, I'll move this to a discussion if that's OK.

**Comment by dtinth** [comment]
With Rosetta 2 it is working perfectly. Been using it for 2 months now.

![jamm1](https://user-images.githubusercontent.com/193136/117109866-83f9cb80-adaf-11eb-8d02-948dc1ddb067.jpg)

**Comment by njc235** [comment]
Have you checked Actuvity Monitor to see if the core audio service is running?

**Comment by mvoitz** [comment]
This is running on my end.

![image](https://user-images.githubusercontent.com/83690768/117140776-11d4b700-adae-11eb-9572-71052a1aaf82.jpeg)

**Reply by mvoitz** [reply]
Error found! Simple! User Error .. as usual! Due to this is a Mac Mini I had not connected my micro in the beging. After it was connected Jamulus runs just perfect! Maybe Jamulus should display a better error message.

**Comment by njc235** [comment]
What had you not connected?  I have a MacMini too.

**Reply by mvoitz** [reply]
My mac mini has no camera (with mic) attached. So the system has no Video and or Mic device. Then I installed the rest of my software and also connected my external sound card from Focusrite and it directly worked. In a way logic but it would be great i Jamulus would have made a more precise error message.

**Comment by njc235** [comment]
Mine will run with no external devices attached as core audio is still there as a system service. Maybe your Mac has a problem?

**Comment by Rockpapst** [comment]
Hello, 

I have a M1 Processor with Big Sur and Jamulus runs very well on a Mac Book Pro (with own server). 

However, I am surprised that unlike the PC variant, there is no menu bar. How can I name my user? 
So I have not found a way yet to not work with public servers as an named user. 

Kind regards
Rockpapst

**Comment by njc235** [comment]
The menu bar is in the usual place in the Mac, not in the window but at the top of the screen. You name the use in the My Profile section under the View menu.

**Comment by Rockpapst** [comment]
Ah ok, thank you very much;-) 
I switched from PC and use mainly Cubase.  
But the M1 rocks.;-) 

Do you have another tip about the latencies? 
I have noticed that I better not use the auto latency.
At least when I run my own server. 

I have a 100 M/Bit fiber line.
What is the optimal setting? 
From time to time it cracks in my (audio)line that I have to adjust the values (latency) upwards. 
With auto function I unfortunately run very fast into a disturbance. 
Hence the question.

**Reply by dcorson-ticino-com** [reply]
Hi Rock,
You will need to try all the possibilities of buffer size and jitter buffer size and see what works best for you.
These problems do not depend on the speed of the Internet connection, but on the regularity (jitter) in which the packets are sent across the Internet.  This is a parameter which the providers don't care much about, they just look at the speed.  It depends also on all the nodes between you and the server.
There is not much one can do to improve the situation except choose a different server which will have a different routing.  Last chance is to try a different provider, but that is often difficult.
While I, for example, have good results with the auto jitter buffer setting, many find going to manual and setting to 2 above the auto setting gives better results.  You have to see for yourself, every Internet connection is different.

**Comment by njc235** [comment]
Make sure your WiFi is off and you are using a wired Ethernet connection to your router. Check that nobody in the house is streaming or gaming for best results.
